safe_transaction_service/history/management/commands/setup_service.py

```diff
@@ -200,28 +225,34 @@ def handle(self, *args, **options):
             self.stdout.write(
                 self.style.SUCCESS(f"Setting up {ethereum_network.name} safe addresses")
             )
-            self._setup_safe_master_copies(MASTER_COPIES[ethereum_network])
+            self._setup_safe_singleton_addresses(MASTER_COPIES[ethereum_network])
+            self._setup_erc20_indexing()
+        else:
+            self._setup_safe_singleton_addresses_from_chain(ethereum_client)
             self._setup_erc20_indexing()
+
```

Comment:
`self._setup_erc20_indexing()` is called in both `if` branches, better call it once after `else` ends